### PR TITLE
express-notes-api - Marianne Cuevas

### PR DESCRIPTION
**Pull request by mariannecuevas**
Screenshot 2023-05-16 200749
Screenshot 2023-05-16 202220
Screenshot 2023-05-18 205117
Screenshot 2023-05-18 205242
Screenshot 2023-05-18 205340
Screenshot 2023-05-18 205523

### CODE REVIEW

File: express-notes-api/server.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    const data = JSON.parse(jsonData);
+
+    data.notes = notesData.notes;
+    data.nextId = notesData.nextId;
+
+    await writeFile('data.json', JSON.stringify(data, null, 2), { encoding: 'utf8' });
+  } catch (err) {
+    console.error(err);
+  }
+}
```
Comment (TimelikeClosure) [inline express-notes-api/server.js]: If a file read or write error occurs during a request, the server should respond with a `500 Internal Server Error`. Please add screenshots to your pull request to verify these responses are sent correctly.

File: express-notes-api/server.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  } catch (err) {
+    console.error(err);
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const id = req.params.id;
+  if (isNaN(id) || id <= 0) {
+    res.status(400).json({ error: 'Id must be a positive integer' });
+  }
```
Comment (TimelikeClosure) [inline express-notes-api/server.js]: Please improve your id validation to send `400 Bad Request` responses on positive non-integer numbers.

File: express-notes-api/server.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  try {
+    const jsonData = await readNotes();
+    const notesData = jsonData.notes;
+    const note = notesData[id];
+
+    if (note) {
+      note.content = req.body.content;
+      await writeNotes(jsonData);
+      res.sendStatus(204);
```
Comment (TimelikeClosure) [inline express-notes-api/server.js]: Successful `PUT`s should result in the correct status code.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by mariannecuevas** [comment]
Screenshot 2023-05-21 173036
Screenshot 2023-05-21 173754
Screenshot 2023-05-21 174039

I'm not quite sure if this is right.

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**